docs/sql-data-sources-text.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: global
+title: Text Files
+displayTitle: Text Files
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+Spark SQL provides `spark.read().text("file_name")` to read a file or directory of files in text format into a Spark DataFrame, and `dataframe.write().text("path")` to write to a text file. Text file does not necessarily mean a specific format ending in `txt`, but literally means any text file even including csv and json. The `option()` function can be used to customize the behavior of reading or writing, such as controlling behavior of the line separator, compression, and so on.
```

Comment:
> Text file does not necessarily mean a specific format ending in `txt` but literally means any text file even ...

This sounds weird. Also we have an encoding option, and it's UTF-8 by default. I would just remove this

Reply:
Thanks for the comment!

Btw, seems like the `encoding` option is not documented anywhere in both [Scala docs](https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/DataFrameReader.html) and [Python docs](https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/DataFrameReader.html).

Reply:
Oh, okay. I forget that text data source doesn't support encoding option. But it reads UTF-8 strings by default in any event.

Reply:
Oh, text doesn't support encoding option. It's just for internal purpose. But it reads UDF-8 by default in any event.